File: Unidad1/Actividad/javascript/ejercicios_records.js
// Declaramos los records
let estudiante1 = { nombre: "Neider", edad: 27, promedio: 4.0 };
let estudiante2 = { nombre: "Valen", edad: 22, promedio: 3.5 };
let estudiante3 = { nombre: "Roosevelt", edad: 33, promedio: 4.4 };

// Guardamos en un arreglo
let estudiantes = [estudiante1, estudiante2, estudiante3];

// Recorremos
console.log("Listado de estudiantes de la Fundación UDEC:");
for (let i = 0; i < estudiantes.length; i++) {
    console.log(
        "Nombre: " + estudiantes[i].nombre +
        ", Edad: " + estudiantes[i].edad +
        ", Promedio: " + estudiantes[i].promedio
    );
}

// Modificamos el promedio, ahora usamos al estudiante1
for (let i = 0; i < estudiantes.length; i++) {
    if (estudiantes[i].nombre === "Neider") {
        estudiantes[i].promedio = 4.5;
    }
}

console.log("\nLa nueva lista de estudiantes es:");
for (let i = 0; i < estudiantes.length; i++) {
    console.log(
        "Nombre: " + estudiantes[i].nombre +
        ", Edad: " + estudiantes[i].edad +
        ", Promedio: " + estudiantes[i].promedio
    );
}

File: Unidad1/Actividad/python/ejercicios_records.py
from dataclasses import dataclass

# Declaramos
@dataclass
class Estudiante:
    nombre: str
    edad: int
    promedio: float

# Creo 3 instancias con datos de mis hermanos
estudiante1 = Estudiante("Neider", 27, 4.0)
estudiante2 = Estudiante("Valen", 22, 3.5)
estudiante3 = Estudiante("Roosevelt", 33, 4.4)

# Guardamos en una lista
estudiantes = [estudiante1, estudiante2, estudiante3]

# Recorremos
print("Listado de estudiantes de la Fundación UDEC:")
for estudiante in estudiantes:
    print(f"Nombre: {estudiante.nombre}, Edad: {estudiante.edad}, Promedio: {estudiante.promedio}")

# Modificamos un promedio
for estudiante in estudiantes:
    if estudiante.nombre == "Valen":
        estudiante.promedio = 4.0

print("\nLa nueva lista de estudiantes es:")
for estudiante in estudiantes:
    print(f"Nombre: {estudiante.nombre}, Edad: {estudiante.edad}, Promedio: {estudiante.promedio}")